pallets/pools/src/lib.rs

```diff
@@ -628,6 +684,42 @@ pub mod pallet {
 				None => None,
 			};
 
+			for (tranche, tranche_input) in tranches.tranches.iter().zip(&tranche_inputs) {
+				let token_name: BoundedVec<u8, T::MaxTokenNameLength> = tranche_input
+					.clone()
+					.metadata
+					.token_name
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
+
+				let token_symbol: BoundedVec<u8, T::MaxTokenSymbolLength> = tranche_input
+					.clone()
```

Comment:
we're cloning `tranche_input` twice here - does it make more sense to do `tranche_input.metadata.token_{name,symbol}.clone()` for each value instead, or to do `let tranche_input = tranche_input.clone()` then extract the values from it?